public/app/features/datasources/state/actions.ts

```diff
@@ -62,7 +62,13 @@ export const initDataSourceSettings = (
       }
 
       const dataSource = dependencies.getDataSource(getState().dataSources, pageId);
-      const dataSourceMeta = dependencies.getDataSourceMeta(getState().dataSources, dataSource.type);
+
+      let dsType = dataSource.type;
+      if (dsType === 'stackdriver') {
```

Comment:
Having this logic here feels like a code smell. Can we handle the legacy aspect from backend solely?